### PR TITLE
[e2e] Fix node port service reachability test for nodes running on localhost

### LINKED ISSUE #80760: network service e2e tests failing
**Which jobs are failing**:
CRI-O e2e

**Which test(s) are failing**:
1. Kubernetes e2e suite [sig-network] Services should be able to change the type from ExternalName to NodePort 
2. Kubernetes e2e suite [sig-network] Services should be able to create a functioning NodePort service 19s

See link: https://openshift-gce-devel.appspot.com/build/origin-federated-results/pr-logs/pull/cri-o_cri-o/2684/test_pull_request_crio_e2e_rhel/5905/

**Since when has it been failing**:
https://github.com/kubernetes/kubernetes/pull/79667

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by saschagrunert**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

If a node's internal address points to 127.0.0.1, then we skip the node
port reachability test.

**Which issue(s) this PR fixes**:

Fixes #80760

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Removing `WIP` since https://github.com/cri-o/cri-o/pull/2707 confirms that it fixes the actual issue.

/cc @mrunalp

**Comment by haircommander** [comment]
/cc @oomichi
/cc @johnbelamaric 
/cc @andrewsykim 
PTAL, CRI-O has been blocked upstream and needs this (copied because you reviewed https://github.com/kubernetes/kubernetes/pull/79667)

**Comment by johnbelamaric** [comment]
Thank you!
/priority critical-urgent 
/sig network
/lgtm 
/approve

File: test/e2e/framework/service/jig.go [hunk 1]
```diff
@@ -773,6 +773,11 @@ func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v
 	internalAddrs := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
 	externalAddrs := e2enode.CollectAddresses(nodes, v1.NodeExternalIP)
 	for _, internalAddr := range internalAddrs {
+		// If the node's internal address points to localhost, then we are not
+		// able to test the service reachability via that address
+		if internalAddr == "127.0.0.1" {
```
Comment (oomichi) [inline test/e2e/framework/service/jig.go]: Don't we need to consider local loopback address of IPv6 here?

File: test/e2e/framework/service/jig.go [hunk 2]
```diff
@@ -773,6 +773,11 @@ func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v
...
+			continue
```
Comment (MrHohn) [inline test/e2e/framework/service/jig.go]: Can we add a log line so it is clear this is skipped?

**Review by MrHohn — COMMENTED** [review]
/lgtm

File: test/e2e/framework/service/jig.go (hunk 1 above)
Comment (andrewsykim) [reply test/e2e/framework/service/jig.go]: [IsLoopback](https://golang.org/src/net/ip.go?s=3230:3260#L108) should do it, but need to `net.ParseIP` on the internal address first

File: test/e2e/framework/service/jig.go (hunk 2 above)
Comment (saschagrunert) [reply test/e2e/framework/service/jig.go]: Yes, added as suggested.

File: test/e2e/framework/service/jig.go (hunk 1 above)
Comment (saschagrunert) [reply test/e2e/framework/service/jig.go]: I added a separate utility function which verifies both.

**Comment by oomichi** [comment]
Thanks for updating

/approve

**Pull request merged**